web_search_with_and/static/src/js/search_bar.js

```diff
@@ -27,7 +27,7 @@ odoo.define("web_search_with_and/static/src/js/search_bar.js", function (require
                     filterId: source.filterId,
                     value:
                         "value" in source
-                            ? source.value.trim()
+                            ? source.value
```

Comment:
Tested on this PR's runboat: added a search field `id` on the res.users search view. Search for value '1'. Same error still occurs.

`source.value` is a string type in line 30, but cast as an integer by `_parseWithSource`. You might want to check the type of the return value of `_parseWithSource` before trimming.

![image](https://github.com/OCA/web/assets/1033124/f5e8cf29-b8f0-41d2-90b8-930cb20455ce)

Reply:
I moved the trim() function inside the _parseWithSource. With that modification, we will never try to trim and integer value and we keep triming the strings.